scrapper.py

```diff
@@ -21,43 +36,265 @@ class IncorrectNumberOfArticlesError(Exception):
+        root_bs = BeautifulSoup(root_response.text, features="html.parser")
+        root_content_bs = root_bs.find('div', {'id': 'content'})
+        root_table_bs = root_content_bs.find_all('table')[8]
+
+        links_bs = root_table_bs.find_all('a')
+
+        seed_urls = []
+
+        for link_bs in links_bs:
+            seed_urls.append(''.join([DOMAIN_URL, link_bs['href']]))
+
+        self._seed_urls = seed_urls
+
+        super().find_articles()
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = pathlib.Path(base_path)
+
+    if path.exists():
+        shutil.rmtree(path)
+
+    path.mkdir(parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r') as crawler_file:
+        crawler_config = json.load(crawler_file)
+
+    if 'total_articles_to_find_and_parse' not in crawler_config:
+        raise IncorrectNumberOfArticlesError
+
+    if 'seed_urls' not in crawler_config:
+        raise IncorrectURLError
+
+    max_articles = crawler_config['total_articles_to_find_and_parse']
+
+    if not isinstance(max_articles, int):
+        raise IncorrectNumberOfArticlesError
+
+    if max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if max_articles > 100:
+        raise NumberOfArticlesOutOfRangeError
+
+    seed_urls = crawler_config['seed_urls']
+
+    if not isinstance(seed_urls, list) or not seed_urls:
+        raise IncorrectURLError
+
+    for seed_url in seed_urls:
+        match = re.match(r'(^http://|^https://)', seed_url)
+
+        if not match or DOMAIN_URL not in seed_url:
+            raise IncorrectURLError
+
+    return seed_urls, max_articles
 
 
 if __name__ == '__main__':
-    # YOUR CODE HERE
-    pass
+    outer_seed_urls, outer_max_articles = validate_config(CRAWLER_CONFIG_PATH)
+
+    try:
+        SCRAPPER_MODE_INPUT = input('Should the environment be reset? Press R. '
+                                    'If you want to continue running the scrapper, press C')
+    except EOFError:
+        SCRAPPER_MODE_INPUT = 'R'
+
+    if SCRAPPER_MODE_INPUT == 'R':
+        prepare_environment(ASSETS_PATH)
+    elif SCRAPPER_MODE_INPUT == 'C':
+        metadata = []
+        assets_pathlib_path = pathlib.Path(ASSETS_PATH)
+
+        for file_name in assets_pathlib_path.iterdir():
+            if file_name.suffix == '.json':
+                with open(file_name, encoding='utf-8') as f:
+                    config = json.load(f)
+
+                stem = file_name.stem
+                number = stem[0]
+
+                pdf_path = ASSETS_PATH / f'{number}_raw.pdf'
+
+                if pdf_path.is_file():
+                    pre_scrapped_urls.append(config['url'])
+    else:
+        print('Incorrect input')
+        raise KeyboardInterrupt
+
+    PRE_SCRAPPED_URL_COUNT = len(pre_scrapped_urls)
```

Comment:
even if I had accepted the global variable for storing the list of URLs, what is the reason for storing its length as a separate variable?